renku/api/models/run.py

```diff
@@ -58,17 +59,30 @@
 )
 from renku.core.plugins.provider import RENKU_ENV_PREFIX
 
+name_pattern = re.compile("[a-zA-Z0-9-_]+")
+
 
 class _PathBase(PathLike):
     @ensure_project_context
-    def __init__(self, path, project):
-        self._path = Path(path)
+    def __init__(self, name: str, path: Union[str, Path], project):
+        if name and not name_pattern.match(name):
```

Comment:
If a name is not provided (or is empty) then thing can go wrong (e.g. multiple inputs will get the same value). So we should also check for that case. Perhaps it's better to put this check in a function since it's also used in another place.